### PR TITLE
[shopsys] deployment

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Ease deployment of the project-base on gitlab. See also https://github.com/shopsys/deployment
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: project-base/app/config/bundles.php [hunk 1]
```diff
@@ -34,7 +34,7 @@
     Symfony\Bundle\WebProfilerBundle\WebProfilerBundle::class => ['dev' => true],
     Symfony\Bundle\DebugBundle\DebugBundle::class => ['dev' => true],
     Symfony\WebpackEncoreBundle\WebpackEncoreBundle::class => ['all' => true],
-    Shopsys\S3Bridge\S3BridgeBundle::class => ['all' => true],
+    Shopsys\S3Bridge\ShopsysS3BridgeBundle::class => ['all' => true],
```
Comment (TomasLudvik) [inline project-base/app/config/bundles.php:37]: Should this be mentioned in UPGRADE or will Symfony update it on its own?

File: project-base/app/app/preload.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+declare(strict_types=1);
+
+if (file_exists(dirname(__DIR__) . '/var/cache/prod/App_KernelProdContainer.preload.php')) {
+    require dirname(__DIR__) . '/var/cache/prod/App_KernelProdContainer.preload.php';
```
Comment (TomasLudvik) [inline project-base/app/app/preload.php:6]: This article (https://symfony.com/blog/php-preloading-and-symfony) says, that you need to add some configuration to php.ini for it to work. Should we add it too?

File: project-base/app/config/packages/frontend_api.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+parameters:
+    shopsys.frontend_api.keys_filepath: '%kernel.project_dir%/config/frontend-api'
+    shopsys.frontend_api.domains:
+        - 1
+        - 2
```
Comment (TomasLudvik) [inline project-base/app/config/packages/frontend_api.yaml:5]: In that case, you should also remove calling `php phing frontend-api-enable` from CI scripts

File: project-base/app/deploy/deploy-project.sh [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        ["AKENEO_BASE_URI"]=${AKENEO_BASE_URI}
+        ["AKENEO_CLIENT_ID"]=${AKENEO_CLIENT_ID}
+        ["AKENEO_SECRET"]=${AKENEO_SECRET}
+        ["AKENEO_USER"]=${AKENEO_USER}
+        ["AKENEO_PASSWORD"]=${AKENEO_PASSWORD}
+
+        ["SSFWCC_BRIDGE_ENABLED"]=${SSFWCC_BRIDGE_ENABLED}
+        ["SSFWCC_BRIDGE_BASE_URI"]=${SSFWCC_BRIDGE_BASE_URI}
+        ["SSFWCC_BRIDGE_USER"]=${SSFWCC_BRIDGE_USER}
+        ["SSFWCC_BRIDGE_PASSWORD"]=${SSFWCC_BRIDGE_PASSWORD}
```
Comment (TomasLudvik) [inline project-base/app/deploy/deploy-project.sh:52]: We should probably update name of these ENV variables to only SSFW

File: project-base/gitlab/mirror-repository/gitlab-stage.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+        - web
+        - triggers
+    needs: []
+    script:
+        - apk add git openssh-client
+        - eval $(ssh-agent -s)
+        - mkdir -p ~/.ssh
+        - '[[ -f /.dockerenv ]] && echo -e "Host *\n\tStrictHostKeyChecking no\n\n" > ~/.ssh/config'
+        - echo "$SSHKEY_MIRROR_PROJECT_BASE" | ssh-add -
+        - sh ./gitlab/mirror-repository/mirror.sh "shopsys-project-base" "git@github.com:shopsys/project-base.git" "13.0" "git@gitlab.shopsys.cz:ss6-projects/project-base.git" "devel"
```
Comment (TomasLudvik) [inline project-base/gitlab/mirror-repository/gitlab-stage.yml:13]: I have a weird feeling about hardcoding 13.0 here - this should be an ENV variable or something.

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

File: project-base/app/app/preload.php (hunk 2 above)
Comment (grossmannmartin) [reply project-base/app/app/preload.php:6]: it's already added – https://github.com/shopsys/deployment/blob/main/kubernetes/configmap/production-php-opcache.yaml

File: project-base/app/config/bundles.php (hunk 1 above)
Comment (grossmannmartin) [reply project-base/app/config/bundles.php:37]: true, added - fdad8c2ef8

File: project-base/app/deploy/deploy-project.sh (hunk 4 above)
Comment (grossmannmartin) [reply project-base/app/deploy/deploy-project.sh:52]: Among other things... cron instance name, whole ssfwcc bridge namespace in components, config in parameters... 
I would make this change separately if you agree.

File: project-base/gitlab/mirror-repository/gitlab-stage.yml (hunk 5 above)
Comment (grossmannmartin) [reply project-base/gitlab/mirror-repository/gitlab-stage.yml]: make sense – changed in 769540a113.
env variable is set in GitLab on CI/CD env variables configuration page

File: project-base/app/deploy/deploy-project.sh (hunk 4 above)
Comment (TomasLudvik) [reply project-base/app/deploy/deploy-project.sh:52]: Ok :)

**Review by TomasLudvik — APPROVED** [review]

**Pull request merged**